### PR TITLE
Handle column sorting events directly on the header cell

### PR DESCRIPTION
**Pull request by nstepien**

### CODE REVIEW

File: src/HeaderCell.tsx [hunk 1]
```diff
@@ -138,15 +157,18 @@ export default function HeaderCell<R, SR>({
     }
   }
 
-  function onClick() {
+  function onClick(event: React.MouseEvent<HTMLSpanElement>) {
+    if (ignoreClickRef.current) return;
```
Comment (nstepien) [inline src/HeaderCell.tsx:161]: Maybe we should just have a real DOM element with `stopPropagation()` on it, would simplify the JS a little bit.

File: website/root.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const mainClassname = css`
   box-sizing: border-box;
   block-size: 100vh;
   padding: 8px;
-  overflow: hidden;
+  overflow: clip;
```
Comment (amanmahajan7) [inline website/root.tsx:36]: Looks like this breaks the navigation style
image

File: src/HeaderCell.tsx (hunk 1 above)
Comment (amanmahajan7) [reply src/HeaderCell.tsx:161]: Yeah. let's do it

File: src/HeaderCell.tsx [hunk 3]
```diff
@@ -62,28 +71,36 @@ export default function HeaderCell<R, SR>({
...
-    const offset = isRtl ? event.clientX - left : right - event.clientX;
+    const offset = getPointerOffset(event);
 
     if (offset > 11) {
       // +1px to account for the border size
       return;
     }
 
+    // prevent selecting/sorting if we started resizing the column, including double-clicking
+    ignoreClickRef.current = true;
```
Comment (amanmahajan7) [inline src/HeaderCell.tsx:98]: We do not use a ref currently. Is there a bug?

Comment (nstepien) [reply src/HeaderCell.tsx:98]: Without it, after resizing the column, the `click` event is triggered on the header cell.
This wasn't an issue before as the `click` event handler was on a child element

**Review by amanmahajan7 — DISMISSED** [review]

**Review by amanmahajan7 — APPROVED** [review]

**Pull request merged**